server/webserver.go

```diff
@@ -378,6 +402,30 @@ func (s *LivepeerServer) cliWebServerHandlers(bindAddr string) *http.ServeMux {
 			return
 		}
 
+		pricePerUnitStr := r.FormValue("pricePerUnit")
+		pricePerUnit, err := strconv.ParseInt(pricePerUnitStr, 10, 64)
+		if err != nil {
+			glog.Errorf("Error converting pricePerUnit string to int64: %v\n", err)
+			return
+		}
+		if pricePerUnit <= 0 {
+			glog.Errorf("price unit must be greater than 0, provided %d\n", pricePerUnit)
+			return
+		}
+
+		pixelsPerUnitStr := r.FormValue("pixelsPerUnit")
+		pixelsPerUnit, err := strconv.ParseInt(pixelsPerUnitStr, 10, 64)
+		if err != nil {
+			glog.Errorf("Error converting pixelsPerUnit string to int64: %v\n", err)
+			return
+		}
+		if pixelsPerUnit <= 0 {
+			glog.Errorf("pixels per unit must be greater than 0, provided %d\n", pixelsPerUnit)
+			return
+		}
+		s.LivepeerNode.PriceInfo = big.NewRat(pricePerUnit, pixelsPerUnit)
```

Comment:
If we do this here, it should be under a lock, and all other accesses to PriceInfo need to be at least read-locked as well.

Reply:
fair point, but then we would need a wrapper `orchestratorConfig` struct. This wouldn't be a big deal but then I feel like we should define it on `LivpeerNode` but on `orchestrator` itself. WDYT?

Reply:
We could just have a mutex in the livepeer node itself rather than introduce yet another wrapper struct. There's some other data that could be protected around the same mutex as well - `BroadcastJobVideoProfiles` comes to mind, although IIRC it lives in `server`. It could also just be a global: https://github.com/livepeer/go-livepeer/pull/879#discussion_r286709412

Writes / reads could also be meditated by a helper function, although IMO golang does not give us convenient misuse resistance due to the lack of const pointer return values.

Reply:
I'd have a preference to not use globals or "package level vars" rather unless they are constants.[ This seems to be an idiom widely spread amongst the go community.
](https://twitter.com/davecheney/status/871939730761547776)

If we use one RWMutex on the node itself, wouldn't we also be locking all values ? What are the side-effects of doing so? 

Another reason I would have no problem with an `orchConfig` wrapper here in this particular case is that it becomes easily extendable and all orchestrater related config lives on that `orchestrator` struct rather than on the universal `LivepeerNode` struct. 

Because currently broadcaster has no use for `node.PriceInfo` as he defines his maxPriceInfo in `server.Broadcastcfg`

Reply:
> If we use one RWMutex on the node itself, wouldn't we also be locking all values ? What are the side-effects of doing so?

We would only lock/unlock the mutex when accessing fields that need to be concurrency safe i.e. `PriceInfo`. I think `serviceURI` also needs to be protected since it can be changed in the CLI.

I would consider eventually introducing a config struct with a single mutex for all fields that are initialized via flags (or parsing a config file) and that can be modified while the node is running if we end up with more of these fields, but I don't think we need to introduce that now since `PriceInfo` already lives within the `LivepeerNode` struct - an additional mutex and helper methods would suffice.

While I agree that this field is O specific and an O specific struct would be a better place for it, `LivepeerNode` holds other O specific state already as well. So, if we wanted to move the field to `orchestrator` we would probably want to move the other O specific state there as well. This feels like a change that doesn't need to happen right now so I would be fine with keeping the field in `LivepeerNode` and then considering a bigger move of O specific state into `orchestrator` separately.

Reply:
Can we use segmentMutex , which is already defined on the node for the time being? 
Creating another general RWMutex on the node seems trivial?

Reply:
`segmentMutex` protects access to `n.SegmentChans` which is specific to the transcoding workflow. I think we should use a separate mutex for protecting config fields like `PriceInfo` - locking/unlocking for a config field shouldn't impact the transcoding workflow.

Reply:
Reinstated segmentMutex for transcoding specific locking/unlocking and created a seperate mutex for other LivepeerNode stuff. Also added locks/unlocks to Set/Get serviceURI 
https://github.com/livepeer/go-livepeer/pull/993/commits/f5edd9c989f6ca0f466973cbc4ad9cf91b166a8c

Reply:
> I would consider eventually introducing a config struct with a single mutex for all fields that are initialized via flags (or parsing a config file) and that can be modified while the node is running

💯 💯 💯 

(The progress here looks good so far; I'll return to the sidelines and leave the rest up to you both.)

Reply:
Thanks for your input Josh, as always !